internal/ir/xds.go

```diff
@@ -2996,6 +2996,8 @@ type DestinationFilters struct {
 	AddResponseHeaders []AddHeader `json:"addResponseHeaders,omitempty" yaml:"addResponseHeaders,omitempty"`
 	// RemoveResponseHeaders defines a list of headers to be removed from response.
 	RemoveResponseHeaders []string `json:"removeResponseHeaders,omitempty" yaml:"removeResponseHeaders,omitempty"`
+	// CredentialInjection defines the credential injection configuration.
+	CredentialInjection *CredentialInjection `json:"credentialInjection,omitempty" yaml:"credentialInjection,omitempty"`
```

Comment:
should the same field at Route level be deleted, and should we push down the IR to this level ? to reuse/consolidate code ?

Reply:
yeah, but I guess the downside is it will create a weighted backend, so lets stick with this